plugins/modules/discovery_intent.py

```diff
@@ -580,105 +629,179 @@ def preprocess_device_discovery_handle_error(self):
         self.log("IP Address list's length is longer than 1", "ERROR")
         self.module.fail_json(msg="IP Address list's length is longer than 1", response=[])
 
-    def http_cred_failure(self, msg=None):
+    def discovery_specific_cred_failure(self, msg=None):
         """
         Method for failing discovery if there is any discrepancy in the http credentials
         passed by the user
         """
+
         self.log(msg, "CRITICAL")
         self.module.fail_json(msg=msg)
 
-    def create_params(self, credential_ids=None, ip_address_list=None):
+    def handle_discovery_specific_credentials(self, new_object_params=None):
         """
-        Create a new parameter object based on the validated configuration,
-        credential IDs, and IP address list.
+        Method to convert values for create_params API when discovery specific paramters
+        are passed as input.
 
         Parameters:
-          - credential_ids: The list of credential IDs to include in the
-                            parameters. If not provided, an empty list is used.
-          - ip_address_list: The list of IP addresses to include in the
-                             parameters. If not provided, None is used.
+            - new_object_params: The dictionary storing various parameters for calling the
+                                 start discovery API
 
         Returns:
-          - new_object_params: A dictionary containing the newly created
-                               parameters.
+            - new_object_params: The dictionary storing various parameters for calling the
+                                 start discovery API in an updated fashion
         """
 
-        if credential_ids is None:
-            credential_ids = []
+        discovery_specific_credentials = self.validated_config[0].get('discovery_specific_credentials')
```

Comment:
if none, can we return back.. I don't see any use of getting other values after this...

All the below assignments also you can do whenever it is needed...